tests/lib/Component/ControllerComponentTest.php

```diff
@@ -0,0 +1,75 @@
+<?php
+
+/**
+ * @copyright Copyright (C) Ibexa AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+declare(strict_types=1);
+
+namespace Ibexa\Tests\TwigComponents\Component;
+
+use Ibexa\TwigComponents\Component\ControllerComponent;
+use PHPUnit\Framework\TestCase;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\RequestStack;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpKernel\HttpKernelInterface;
+
+final class ControllerComponentTest extends TestCase
+{
+    public function testRenderReturnsSubRequestResponseContent(): void
+    {
+        $expectedOutput = 'Rendered content';
+
+        $request = new Request();
+        $requestStack = $this->createMock(RequestStack::class);
+        $requestStack
+            ->method('getCurrentRequest')
+            ->willReturn($request);
+
+        $response = new Response($expectedOutput);
+
+        $httpKernel = $this->createMock(HttpKernelInterface::class);
+        $httpKernel
+            ->expects(self::once())
+            ->method('handle')
+            ->with(self::callback(function (Request $subRequest): bool {
```

Comment:
Why not:
```suggestion
            ->with(self::callback(function (Request $subRequest): void {
```

Reply:
from what I can see  `self::callback` callable parameter needs to return `bool`